packages/core/src/util/runtime.ts

```diff
@@ -177,3 +180,22 @@ export const isEnabled = (
 
   return true;
 };
+
+export const isInherentlyEnabled = (
+  state: JsonFormsState,
+  ownProps: any,
+  uischema: UISchemaElement,
+  schema: JsonSchema & { readonly?: boolean; readOnly?: boolean },
+  rootData: any,
+  config: any
+) => {
+  const readonly = [state.jsonforms, uischema.options, schema, config].some(
+    (prop: any) => prop?.readonly || prop?.readOnly
+  );
+  const enabled: boolean =
+    !readonly &&
+    (ownProps.enabled === undefined || hasEnableRule(uischema)
+      ? isEnabled(uischema, rootData, ownProps.path, getAjv(state))
+      : ownProps.enabled);
+  return enabled;
+};
```

Comment:
I would prefer when this determination would exactly follow the algorithm presented in #1735, i.e.

```ts
if (state.readonly) {
  return false
}
if (hasEnableRule(uischema)) {
  return isEnabled(uischema, rootData, ownProps.path, getAjv(state));
}
if (typeof uischema.options.readonly === 'boolean') {
  return uischema.options.readonly
}
// ...
if (typeof ownProps.enabled === 'boolean) {
  return ownProps.enabled;
}
return true;
```

While the code might not look nice, the determination can be very easily followed ;)
The idea of this algorithm is not to `||` all (most) potential readonlys but to stop once a value for it was given on purpose. 

Is this fine for you or do you need to `OR` everything for your use cases?

Reply:
got it will update it accordingly. I assumed OR would do :)